src/Raven.Server/Documents/DocumentsStorage.cs

```diff
@@ -1129,6 +1129,18 @@ public IReadOnlyList<ChangeVectorEntry[]> DeleteConflictsFor(DocumentsOperationC
                     // will not detect a conflict. It is an optimization only that
                     // we have to do, so we'll handle it.
 
+                    //Only register the event if we actually deleted any conflicts
+                    if (list.Count > 0)
+                    {
+                        context.Transaction.InnerTransaction.LowLevelTransaction
+                                .AfterCommitWhenNewReadTransactionsPrevented +=
+                            () =>
+                            {
+                                //I don't do this as an interlocked operation on purpose since interlock is expensive and 
+                                //we don't mind getting false positives once in a while.
+                                _documentDatabase.DocumentsStorage._hasConflicts -= list.Count;
```

Comment:
You actually do care, another write transaction may also be doing this, which will lead to unexpected values.
Note that this is actually pretty rare thing to have, and we are fine paying the perf for that.